src/rgw/store/dbstore/dbstore_mgr.cc

```diff
@@ -2,6 +2,13 @@
 // vim: ts=8 sw=2 smarttab
 
 #include "dbstore_mgr.h"
+#include <boost/lockfree/queue.hpp>
```

Comment:
I don't have experience with boost::lockfree:::queue specifically, but a lot with other ("disruptor"/ring-buffer style) lockfree queues;  I've heard complaints;  make sure you are not deep copying (use a queue of T*), see https://stackoverflow.com/questions/43540943/using-boost-lockfree-queue-is-slower-than-using-mutexes

Reply:
fwiw, we used the mpmc fixed queue from https://github.com/mstump/queues in cohort--it appeared to work well (delivered over 2M iops in practice--the 54M too, running the (perfectly balanced enc/deq) test case

Reply:
(not claiming this is better, either, just making a note)